protocol/x/clob/keeper/process_operations.go

```diff
@@ -675,6 +675,25 @@ func (k Keeper) PersistMatchDeleveragingToState(
 	}
 	deltaBaseQuantumsIsNegative := position.GetIsLong()
 
+	// If there are zero-fill deleveraging operations, this is a sentinel value to indicate a subaccount could not be
+	// liquidated or deleveraged and still has negative equity. Mark the current block number in state to indicate a
+	// negative TNC subaccount was seen.
+	if len(matchDeleveraging.GetFills()) == 0 {
+		if !shouldDeleverageAtBankruptcyPrice {
+			panic(
+				fmt.Sprintf(
+					"PersistMatchDeleveragingToState: zero-fill deleveraging operation included for subaccount %+v"+
+						" and perpetual %d but cannot be be deleveraged at bankruptcy price",
+					liquidatedSubaccountId,
+					perpetualId,
+				),
+			)
```

Comment:
Is there a reason we are panicking here instead of throwing an error?

Does it make sense to incr a counter for the error to track how often this is happening?
Does it make sense to add telemetry for the SetNegativeTncSubaccountSeenAtBlock fn so we know how often the success (valid zero fill deleverage) case happens?

Reply:
Will update this to throw an error + add telemetry